File: app/src/main/java/com/example/getfood/ui/loginregister/registerfragment/RegisterPresenter.java
package com.example.getfood.ui.loginregister.registerfragment;

import androidx.annotation.NonNull;
import android.util.Log;

import com.example.getfood.ui.base.BasePresenter;
import com.example.getfood.utils.AppUtils;
import com.fazemeright.firebase_api__library.listeners.OnTaskCompleteListener;

public class RegisterPresenter<V extends RegisterMvpView> extends BasePresenter<V> implements RegisterMvpPresenter<V> {

    public RegisterPresenter() {
    }

    @Override
    public void performRegistration(@NonNull final String userEmail, @NonNull final String password, String confirmPassword) {
        //Validating all entries First

        if (!AppUtils.isEmailValid(userEmail)) {
            getMvpView().valueEntryError("Please enter a valid Nirma University Email Address");
            return;
        }

        if (!password.equals(confirmPassword)) {
            getMvpView().valueEntryError("Both passwords should match");
            return;
        }

        if (!AppUtils.isValidPassword(password)) {
            getMvpView().valueEntryError("Please enter a valid password more than 8 characters");
            return;
        }

        apiManager.createNewUserWithEmailPassword(userEmail, password, new OnTaskCompleteListener() {
            @Override
            public void onTaskSuccessful() {
                Log.d("##DebugData", String.valueOf(apiManager.isUserEmailVerified()));
                Log.d("##DebugData", apiManager.getCurrentUserEmail());
                getMvpView().onUserCreatedSuccessfully();
            }

            @Override
            public void onTaskCompleteButFailed(String errMsg) {
                getMvpView().valueEntryError(errMsg);
            }

            @Override
            public void onTaskFailed(Exception e) {

            }
        });
    }

    @Override
    public void verifyUserEmail() {
        if (apiManager.isUserEmailVerified()) {
            getMvpView().userVerifiedSuccessfully();
        } else {
            apiManager.sendEmailVerification(new OnTaskCompleteListener() {
                @Override
                public void onTaskSuccessful() {
                    getMvpView().userVerifiedSuccessfully();
                }

                @Override
                public void onTaskCompleteButFailed(String errMsg) {
                    getMvpView().valueEntryError(errMsg);
                }

                @Override
                public void onTaskFailed(Exception e) {
                    getMvpView().onUserEmailVerificationFailed();
                }
            });
        }
    }

    @Override
    public boolean isTermsAndConditionChecked(boolean checked) {
        return checked;
    }

    @Override
    public void signOutUser() {
        apiManager.forceSignOutUser();
    }
}

File: app/src/main/java/com/example/getfood/ui/loginregister/registerfragment/RegisterPresenter.kt
package com.example.getfood.ui.loginregister.registerfragment

import android.util.Log
import com.example.firebase_api_library.listeners.OnTaskCompleteListener
import com.example.getfood.ui.base.BasePresenter
import com.example.getfood.utils.AppUtils

class RegisterPresenter<V : RegisterMvpView?> : BasePresenter<V>(), RegisterMvpPresenter<V> {
    override fun performRegistration(userEmail: String, password: String, confirmPassword: String) {
        //Validating all entries First
        if (!AppUtils.isEmailValid(userEmail)) {
            mvpView?.valueEntryError("Please enter a valid Nirma University Email Address")
            return
        }
        if (!(password == confirmPassword)) {
            mvpView?.valueEntryError("Both passwords should match")
            return
        }
        if (!AppUtils.isValidPassword(password)) {
            mvpView?.valueEntryError("Please enter a valid password more than 8 characters")
            return
        }
        apiManager!!.createNewUserWithEmailPassword(userEmail, password, object : OnTaskCompleteListener {
            override fun onTaskSuccessful() {
                Log.d("##DebugData", apiManager!!.isUserEmailVerified.toString())
                Log.d("##DebugData", "${apiManager?.userEmail}")
                mvpView?.onUserCreatedSuccessfully()
            }

            override fun onTaskCompleteButFailed(errMsg: String?) {
                mvpView?.valueEntryError(errMsg)
            }

            override fun onTaskFailed(e: Exception?) {}
        })
    }

    override fun verifyUserEmail() {
        if (apiManager!!.isUserEmailVerified) {
            mvpView?.userVerifiedSuccessfully()
        } else {
            apiManager!!.sendEmailVerification(object : OnTaskCompleteListener {
                override fun onTaskSuccessful() {
                    mvpView?.userVerifiedSuccessfully()
                }

                override fun onTaskCompleteButFailed(errMsg: String?) {
                    mvpView?.valueEntryError(errMsg)
                }

                override fun onTaskFailed(e: Exception?) {
                    mvpView?.onUserEmailVerificationFailed()
                }
            })
        }
    }

    override fun isTermsAndConditionChecked(checked: Boolean): Boolean {
        return checked
    }

    override fun signOutUser() {
        apiManager!!.forceSignOutUser()
    }
}